### PR TITLE
Additional implementation of UART module

### PR DESCRIPTION
**Pull request by sae-bom**
- Implement UART for Nuttx
- Support opening multiple ports for UART
- Support longer data as an argument of UART.write()
- Update documents about UART with Nuttx feature

IoT.js-DCO-1.0-Signed-off-by: Saebom Kim sae-bom.kim@samsung.com

### CODE REVIEW

File: docs/IoT.js-API-UART.md [hunk 1]
```diff
@@ -6,13 +6,14 @@ The following shows uart module APIs available for each platform.
 
 |  | Linux<br/>(Ubuntu) | Raspbian<br/>(Raspberry Pi) | Nuttx<br/>(STM32F4-Discovery) |
 | :---: | :---: | :---: | :---: |
-| uart.open | O | O | X |
-| uart.close | O | O | X |
-| uart.write | O | O | X |
+| uart.open | O | O | O |
+| uart.close | O | O | O |
+| uart.write | O | O | O |
```
Comment (glistening) [inline docs/IoT.js-API-UART.md]: In first look, I thought it is strange since there is no read method. What about including 'data' event in this table?

Comment (sae-bom) [inline docs/IoT.js-API-UART.md]: In the docs for I2C and net.Socket modules, a description of the 'data' event listener is provided below, apart from the table. I wanted to sync with them.

Comment (glistening) [inline docs/IoT.js-API-UART.md]: OK. I see. Let's determine after offline discuss.

File: docs/IoT.js-API-UART.md [hunk 2]
```diff
@@ -81,3 +85,27 @@ Closes UART device.
 * `callback: Function(err)`
 
 Writes data to UART device.
+
+
+## Particular information for Nuttx
+
+To use UART on stm32f4-discovery board, you must use proper pins. Stm32f4-discovery board supports 4 UART ports, such as USART2, USART3, UART5, USART6.
```
Comment (hs0225) [inline docs/IoT.js-API-UART.md]: As I know, there are USART1, USART4 on stm32f4-discovery board. Why don't you support them? Do you need to modify configure source file on nuttx?

File: src/platform/iotjs_module_uart-posix-general.inl.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+#include <errno.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+#include "module/iotjs_module_uart.h"
+
+static iotjs_jval_t jthis;
+static int fd;
+static uv_poll_t poll_handle;
```
Comment (hs0225) [inline src/platform/iotjs_module_uart-posix-general.inl.h]: On nuttx, there are many UART. If user creates more than one UART, this global variables make a risk of collision.

Comment (sae-bom) [inline src/platform/iotjs_module_uart-posix-general.inl.h]: You're right! I'll try to find another way.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**